doc/crypto/api/keys/policy.rst

```diff
@@ -302,3 +316,39 @@ The usage flags are encoded in a bitmask, which has the type `psa_key_usage_t`.
     .. admonition:: Implementation note
 
         This is a simple accessor function that is not required to validate its inputs. It can be efficiently implemented as a ``static inline`` function or a function-like-macro.
+
+.. function::  psa_check_key_usage
+
+    .. summary::
+        Queries the capabilities of a PSA key object. 
+
+        .. versionadded:: 1.4
+
+    .. param:: psa_key_id_t key
+        a PSA key identifier.
+
+    .. param:: psa_algorithm_t alg
+        a specific algorithm. 
+
+    .. param:: psa_key_usage_t usage
+         a single PSA_KEY_USAGE_xxx flag. 
+
+    .. return:: psa_status_t
+
+    If the supplied key is a key pair, the function checks the appropriate half of the key pair. For example, if the usage flag was `PSA_KEY_USAGE_SIGN_MESSAGE`, it would check the private key. But if it were `PSA_KEY_USAGE_VERIFY_MESSAGE` it would check the public key. 
```

Comment:
The first thing the description should explain is that this function checks multiple things:

* What is technically possible: a positive response indicates that the key is meaningful for the specified algorithm and usage. For example, a public key will never be acceptable for `PSA_KEY_USAGE_SIGN_HASH`, regardless of policy.
* What is permitted by policy. This one is mostly straightforward, except for `PSA_KEY_USAGE_DERIVE_PUBLIC`.
* Any implementation restriction. For example, an implementation that accepts small RSA keys for decryption but not for signature generation should answer yes or no based on that restriction, with the same criteria as its implementation of `psa_asymmetric_decrypt` vs `psa_sign_*`.